java/java.mx.project/src/org/netbeans/modules/java/mx/project/SuiteSources.java

```diff
@@ -0,0 +1,1195 @@
+            }
+        }
+        return null;
+    }
+
+    @Override
+    public SourceForBinaryQuery.Result findSourceRoots(URL url) {
+        return findSourceRoots2(url);
+    }
+
+    final Iterable<File> jdks() {
+        Set<File> jdks = new LinkedHashSet<>();
+        String home = System.getProperty("user.home");
+        if (home != null) {
+            File userEnv = new File(new File(new File(home), ".mx"), "env");
+            findJdksInEnv(jdks, userEnv);
+        }
+        FileObject suiteEnv = dir.getFileObject("mx." + dir.getNameExt() + "/env");
+        if (suiteEnv != null) {
+            findJdksInEnv(jdks, FileUtil.toFile(suiteEnv));
+        }
+
+        String javaHomeEnv = System.getenv("JAVA_HOME");
+        if (javaHomeEnv != null) {
+            jdks.add(new File(javaHomeEnv));
+        }
+        String javaHomeProp = System.getProperty("java.home");
+        if (javaHomeProp != null) {
+            jdks.add(new File(javaHomeProp));
+        }
+        return jdks;
+    }
+
+    private void findJdksInEnv(Set<File> jdks, File env) {
+        if (env == null || !env.isFile()) {
+            return;
+        }
+        try (final FileInputStream is = new FileInputStream(env)) {
+            Properties p = new Properties();
+            p.load(is);
+
+            String javaHome = p.getProperty("JAVA_HOME");
+            if (javaHome != null) {
+                jdks.add(new File(javaHome));
+            }
+
+            String extraJavaHomes = p.getProperty("EXTRA_JAVA_HOMES");
+            if (extraJavaHomes != null) {
+                for (String extraHome : extraJavaHomes.split(File.pathSeparator)) {
+                    jdks.add(new File(extraHome));
+                }
+            }
+        } catch (IOException ex) {
+            Exceptions.printStackTrace(ex);
+        }
+    }
+
+    @Override
+    public SourceLevelQueryImplementation2.Result getSourceLevel(FileObject fo) {
+        Group g = findGroup(fo);
+        if (g == null) {
+            return null;
+        }
+        return new SourceLevelQueryImplementation2.Result2() {
+            @Override
+            public SourceLevelQuery.Profile getProfile() {
+                return SourceLevelQuery.Profile.DEFAULT;
+            }
+
+            @Override
+            public String getSourceLevel() {
+                return g.getCompliance().getSourceLevel();
+            }
+
+            @Override
+            public void addChangeListener(ChangeListener cl) {
+            }
+
+            @Override
+            public void removeChangeListener(ChangeListener cl) {
+            }
+        };
+    }
+
+    @Override
+    public Set<? extends Project> getSubprojects() {
+        Set<Project> prjs = new HashSet<>();
+        for (SuiteSources imp : imported.values()) {
+            prjs.add(imp.prj);
+        }
+        return prjs;
+    }
+
+    @Override
+    public URL[] findUnitTests(FileObject fo) {
+        return new URL[0];
```

Comment:
no unit tests reported ?

Reply:
This is one of the reasons why special `mx` support is desirable. Typical NetBeans projects assume two kinds of directories - main and test. This is not true for `mx` there is no association of main code and test code. Tests are regular source roots with classes annotated by `@Test`. As such this code returns `new URL[0]`.

Feel free to improve it, if you know how.